flink-runtime/src/main/java/org/apache/flink/runtime/io/network/partition/ResultSubpartition.java

```diff
@@ -115,52 +115,62 @@ protected Throwable getFailureCause() {
 
 	public abstract boolean isReleased();
 
+	public abstract boolean isFinished();
```

Comment:
Does it have to be public? If yes, then I think it should be named `isFinishedUnsafe()`. More or less we are creating a convention here as we go, but I think for private/protected methods we can ignore unsafe suffixes as long as there exists only one version of such method. Here existence of thread-safe `isReleased()` adds to the confusion and suggests that `isFinished()` should be thread-safe as well.

Reply:
The access modifiers for these two methods could be narrowed down to `protected`. And I agree with ignoring the unsafe suffixes for them after changing to `protected`.